File: lanjii/lanjii-admin-ui/src/types/sys/sysUser.ts
/**
 * 系统用户实体
 */
export interface SysUser {
    /** 租户ID */
    tenantId?: number
    /** 用户ID */
    id?: number
    /** 用户名 */
    username?: string
    /** 密码 */
    password?: string
    /** 昵称 */
    nickname?: string
    /** 邮箱 */
    email?: string
    /** 手机号 */
    phone?: string
    /** 头像URL */
    avatar?: string
    /** 是否启用（1启用 0禁用） */
    isEnabled?: number
    /** 所属部门 ID */
    deptId?: number
    /** 最后登录时间 */
    lastLoginTime?: string
    /** 最后登录IP */
    lastLoginIp?: string
    /** 是否系统管理员（0-否，1-是） */
    isAdmin?: number
    /** 创建时间 */
    createTime?: string
    /** 更新时间 */
    updateTime?: string
    /** 创建人 */
    createBy?: string
    /** 更新人 */
    updateBy?: string
    /** 删除标志（0-未删除，1-已删除） */
    deleted?: number
}

File: lanjii/lanjii-modules/module-system/system-biz/src/main/java/com/lanjii/sys/entity/SysUser.java
package com.lanjii.sys.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.lanjii.system.api.dto.SysUserDTO;
import com.lanjii.system.api.vo.SysUserVO;
import com.lanjii.framework.mp.base.TenantBaseEntity;
import com.lanjii.sys.base.AdminBaseMapper;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.factory.Mappers;

import java.util.Date;

/**
 * 系统用户表(SysUser) 实体类
 *
 * @author lanjii
 */
@EqualsAndHashCode(callSuper = true)
@Data
@TableName("sys_user")
public class SysUser extends TenantBaseEntity {

    /**
     * 用户ID
     */
    private Long id;

    /**
     * 用户名
     */
    private String username;

    /**
     * 密码
     */
    private String password;

    /**
     * 昵称
     */
    private String nickname;

    /**
     * 邮箱
     */
    private String email;

    /**
     * 手机号
     */
    private String phone;

    /**
     * 头像URL
     */
    private String avatar;

    /**
     * 是否启用（1启用 0禁用）
     */
    private Integer isEnabled;

    /**
     * 所属部门 ID
     */
    private Long deptId;

    /**
     * 最后登录时间
     */
    private Date lastLoginTime;

    /**
     * 最后登录IP
     */
    private String lastLoginIp;

    /**
     * 是否系统管理员（0-否，1-是）
     */
    private Integer isAdmin;

    @Mapper
    public interface SysUserMapper extends AdminBaseMapper<SysUser, SysUserVO, SysUserDTO> {

        @Mapping(target = "deptName",expression = "java(deptIdToName(entity.getDeptId()))")
        SysUserVO toVo(SysUser entity);

    }

    public static final SysUserMapper INSTANCE = Mappers.getMapper(SysUserMapper.class);

}
